### PR TITLE
Prepared release 3.1.0

### LINKED ISSUE #330: 📦 AI-Lab Release 3.1.0
* [x] Adjust version number in file `pyproject.toml`
* [ ] Create release 0.3.0 of notebook-connector
* [x] Update dependency to notebook connector

If you decided to change the version number then
* [x] Run `poetry run version-check <path/version.py> --fix` (only when using PTB)
* [x] Rename latest file `doc/changes/changes_*.md` accordingly
* [x] Update reference in file `doc/changes/changelog.md`
* [x] Update version number in latest file `doc/changes/changes_*.md`

Always
* latest file `doc/changes/changes_*.md`
  * [x] Append release date to first line
  * [x] Add a code name and a summary
* Update version number in user guide
  * [x] `user_guide/docker/docker-usage.md`
  * [x] `user_guide/vm-edition/win-vbox.md`
* Update version number in developer guide
  * [x] `testing.md`
* Replace dependencies to unreleased version (e.g. git dependencies) in files, see instructions in [notebooks.md](https://github.com/exasol/ai-lab/blob/main/doc/developer_guide/notebooks.md):
  * [x] [notebook_requirements.txt](https://github.com/exasol/ai-lab/blob/main/exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook_requirements.txt)
  * [x] [jupyter_requirements.txt](https://github.com/exasol/ai-lab/blob/main/exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/jupyter_requirements.txt)
  * [x] [pyproject.toml](https://github.com/exasol/ai-lab/blob/main/pyproject.toml)

Ship the Actual Release
* [ ] Run release droid `java -jar ~/java/jar/release-droid-*.jar -n ai-lab --goal release`
* [ ] Edit the Release on GitHub making the draft final

Post-Release Actions
* [ ] Update links in file `README.md` to point to the tag on GitHub

### PR DESCRIPTION
**Pull request by ckunki**
Closes #330

### CODE REVIEW

File: doc/changes/changes_3.1.0.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
-# AI-Lab 3.1.0 released T.B.D.
+# AI-Lab 3.1.0 released 2024-09-09
```
Comment (kaklakariada) [inline doc/changes/changes_3.1.0.md:1]:
```suggestion
# AI-Lab 3.1.0 released 2024-09-10
```

File: doc/changes/changes_3.1.0.md [hunk 2]
```diff
@@ -1,4 +1,10 @@
...
+
+Code name: Additional fixes on top of release 3.0.0
+
+## Summary
+
+This release applies additional fixes on top of release 3.0.0.
```
Comment (kaklakariada) [inline doc/changes/changes_3.1.0.md:7]: This is a bit vague, but OK ;)

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook_requirements.txt [hunk 3]
```diff
@@ -2,7 +2,7 @@ scikit-learn==1.5.1 # required for notebook sklearn
 matplotlib==3.7.4 # required for notebook sklearn
 jupysql==0.10.10 # required for multiple notebooks
 stopwatch.py>=2.0.1 # also required by ITDE
-exasol-notebook-connector @ git+https://github.com/exasol/notebook-connector@63a755a
+exasol-notebook-connector>=0.3.0
```
Comment (tkilias) [inline exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook_requirements.txt:5]: I think, this should be == instead

Comment (ckunki) [reply exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook_requirements.txt:5]: oops - definitely! Thanks!

**Review by tkilias — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**